data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -148,19 +155,19 @@ private static DataServiceException createExceptionFromResponse(String url, Http
              *   result), or (2) in the KustoOperationResult's QueryCompletionInformation, both of which present with "200 OK". See .Net's DataReaderParser.
              */
             String activityId = determineActivityId(httpResponse);
-            if (StringUtils.isBlank(errorFromResponse)) {
-                errorFromResponse = String.format("Http StatusCode='%s', ActivityId='%s'", httpResponse.getStatusLine().toString(), activityId);
-                return new DataServiceException(url, errorFromResponse, thrownException, false);
-            } else {
+            if (!StringUtils.isBlank(errorFromResponse)) {
                 String message = "";
                 DataWebException formattedException = new DataWebException(errorFromResponse, httpResponse);
                 try {
                     message = String.format("%s, ActivityId='%s'", formattedException.getApiError().getDescription(), activityId);
+                    return new DataServiceException(url, message, formattedException,
+                            formattedException.getApiError().isPermanent());
                 } catch (Exception ignored) {
                 }
-                return new DataServiceException(url, message, formattedException,
-                        formattedException.getApiError().isPermanent());
             }
+            errorFromResponse = String.format("Http StatusCode='%s', ActivityId='%s'", httpResponse.getStatusLine().toString(), activityId);
```

Comment:
This may overwrite errorFromResponse's value.

I'd put this 1 line in an else.

Reply:
Sorry, I don't follow.

We can wrap just this 1 line in an else, and the following line - the return, will be called no matter what.

Also, note that the catch on line 165 above does nothing and the execution passes to line 168.